core/tests/protractor_utils/waitFor.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utilities for delaying actions with Protractor's
+ * ExpectedConditions.
+ */
+
+var until = protractor.ExpectedConditions;
+var waitTime = 5000;
+
+/**
+ * @param {string} errorMessage - Error message when alert does not pop up.
+ * @param {int} [waitTime] - Optional wait time (in milliseconds) parameter.
```

Comment:
Let's drop this here and elsewhere, if that's OK? We can introduce it later if needed but I think it's better to start without it (to try and more strongly encourage standardization).

Reply:
If I recalled correctly, I used wait time bigger than 5000ms for a few scenarios:

1)First time entering the exploration editor and wait for tutorial modal to animate. 

2)After skipping/playing the tutorial in the editor (wait for "Click here for help in future" bubble to disappear)

3) Flaky actions such as setStateContent or setStateName, this can be due to the help bubble not fully dismissed yet the textarea element is clickable.

I am ok with dropping the waitTime parameter here. If the suite breaks down, that means at that point, an expected condition needs to be put in place.

Reply:
Ah I see. So typically all expected conditions themselves happen within 5 secs (at least, the ones you observed)?

Reply:
Yes! The ones that needs more than 5 secs need to be investigated further.